src/Engine/Objects/NPC.h

```diff
@@ -98,15 +98,14 @@ struct NPCStats {
     char *sub_495366_MispronounceName(uint8_t firstLetter,
                                       uint8_t genderId);
 
-    NPCData pNPCData[501]{};     // 0 - 94BCh count from 1
-    NPCData pNewNPCData[501]{};  // 94BCh- 12978h count from 1
+    std::array<NPCData, 501> pNPCData;     // 0 - 94BCh count from 1
```

Comment:
There are a few points where index overflow was used to access pNewNPCData from this now causing an error.
L89 NPC.cpp
L2009 UIGame.cpp
Please could you address?

Reply:
Thanks, fixed these